tests/unit_tests/device_setup_plans/test_utils.py

```diff
@@ -1,27 +1,38 @@
 from unittest.mock import MagicMock
 
 import pytest
+from bluesky import FailedStatus
 from bluesky import plan_stubs as bps
 from bluesky.run_engine import RunEngine
 from dodal.beamlines import i03
+from ophyd.status import Status
 
 from hyperion.device_setup_plans.utils import (
     start_preparing_data_collection_then_do_plan,
 )
 
 
-def test_given_plan_raises_when_exception_raised_then_eiger_disarmed_and_correct_exception_returned():
-    class TestException(Exception):
-        pass
+@pytest.fixture()
+def mock_eiger():
+    eiger = i03.eiger(fake_with_ophyd_sim=True)
+    eiger.detector_params = MagicMock()
+    eiger.async_stage = MagicMock()
+    eiger.disarm_detector = MagicMock()
+    return eiger
+
 
+class TestException(Exception):
+    pass
+
+
+def test_given_plan_raises_when_exception_raised_then_eiger_disarmed_and_correct_exception_returned(
+    mock_eiger,
+):
     def my_plan():
         yield from bps.null()
         raise TestException()
 
-    eiger = i03.eiger(fake_with_ophyd_sim=True)
-    eiger.detector_params = MagicMock()
-    eiger.async_stage = MagicMock()
-    eiger.disarm_detector = MagicMock()
+    eiger = mock_eiger
     detector_motion = MagicMock()
 
     RE = RunEngine()
```

Comment:
To make tests cleanup more reliably I think it's best if we always use the `RunEngine` fixture in `tests/conftest.py`